### PR TITLE
Various bug fixes and usability improvements

### PR DESCRIPTION
**Pull request by stefannica**
## Describe changes

This is a collection of fixes and usability improvements revealed by the documentation efforts:

* allow the active stack to be overridden through an environment variable (previously not possible)
* remove the SQLAlchemy warnings in the SQL ZenStore
* add the Kubeflow hostname attr to the Kubeflow orchestrator
* various ZenServer improvements:

  * allow to configure the IP address where the server accepts connections
  * store all ZenServer logs and config files in the global config path
  instead of a temp directory
  * make `zenml server` CLI commands less confusing by removing the
  "suspend" feature
  * pass the profile name directly to the uvicorn app and remove the
  code that passes the serialized Profile configuration
  * fix the global configuration unit tests that was breaking the
  os.environ contents and update zen server unit tests to use profiles

## Pre-requisites
Please ensure you have done the following:
- [x] I have read the **CONTRIBUTING.md** document.
- [x] If my change requires a change to docs, I have updated the documentation accordingly.
- [ ] If I have added an integration, I have updated the [integrations](https://docs.zenml.io/features/integrations) table.
- [x] I have added tests to cover my changes.

## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Other (add details above)

### CODE REVIEW

File: src/zenml/cli/server.py [hunk 1]
```diff
@@ -55,60 +60,62 @@ def explain_server() -> None:
...
-def up_server(port: int, profile: Optional[str]) -> None:
+def up_server(ip_address: str, port: int, profile: Optional[str]) -> None:
     """Provisions resources for the ZenServer."""
     from zenml.services import ServiceRegistry
     from zenml.zen_server.zen_server import ZenServer, ZenServerConfig
 
+    service_config = ZenServerConfig(
+        root_runtime_path=GLOBAL_ZENML_SERVER_CONFIG_PATH,
+        singleton=True,
+        ip_address=str(ip_address),
```
Comment (AlexejPenner) [inline src/zenml/cli/server.py:76]: Typing in the signature says ip_address: str - how come you need to cast to str here?

**Review by AlexejPenner — APPROVED** [review]
A bunch of cool little improvements. Have two tiny remarks, but happy to pre-approve as they are very tiny :)

File: src/zenml/zen_server/zen_server.py [hunk 2]
```diff
@@ -121,7 +124,16 @@ def __init__(
         super().__init__(config=config, **attrs)
 
     def run(self) -> None:
-        if self.config.store_profile_configuration.store_type == StoreType.REST:
+        profile = GlobalConfiguration().get_profile(self.config.profile_name)
+        print(
```
Comment (schustmi) [inline src/zenml/zen_server/zen_server.py:128]: this should be `logger.info(...)` I think

**Review by schustmi — CHANGES_REQUESTED** [review]
One super tiny CR, otherwise looks great!

File: src/zenml/cli/server.py [hunk 3]
```diff
@@ -136,49 +143,17 @@ def status_server() -> None:
...
+        cli_utils.error("No ZenServer instance found locally!")
     else:
-        if force:
-            cli_utils.declare(
-                f"Deprovisioning resources for service "
-                f"'{zervice.SERVICE_TYPE.name}'."
-            )
-            zervice.stop()
+        cli_utils.declare("Shutting down the local ZenService instance.")
+        zervice.stop()
```
[the remark this answers is not in the document]
Comment (stefannica) [reply src/zenml/cli/server.py:157]: I don't know too much about this term, I'm just reusing the code that was already there :) Besides, I consider this to be one of those harmless things that put a smile on your face when you see them in the code.

File: src/zenml/cli/server.py (hunk 1 above)
Comment (stefannica) [reply src/zenml/cli/server.py:76]: You're correct. Mypy doesn't seem to catch these, as they are auto-generated by click anyway, but this is an IPv4/v6 address.

File: src/zenml/zen_server/zen_server.py (hunk 2 above)
Comment (stefannica) [reply src/zenml/zen_server/zen_server.py:128]: I had a long unit-test debugging session and this must've gotten away from me. I'll remove it. Thank you for spotting it !

File: src/zenml/cli/server.py [hunk 4]
```diff
@@ -55,60 +61,68 @@ def explain_server() -> None:
...
+def up_server(
+    ip_address: Union[ipaddress.IPv4Address, ipaddress.IPv6Address],
+    port: int,
+    profile: Optional[str],
+) -> None:
     """Provisions resources for the ZenServer."""
     from zenml.services import ServiceRegistry
     from zenml.zen_server.zen_server import ZenServer, ZenServerConfig
 
+    print(type(ip_address))
```
Comment (schustmi) [inline src/zenml/cli/server.py:78]: One `print` gone, another one appeared 😄

Comment (stefannica) [reply src/zenml/cli/server.py]: OMG, this is embarassing :)

**Review by schustmi — APPROVED** [review]
LGTM!

**Pull request merged**